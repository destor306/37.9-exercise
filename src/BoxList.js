import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import {v4 as uuid} from "uuid"

const BoxList = ()=>{
    
    const [boxes, setBoxes] = useState([]);
    const addBox = (newBox) =>{
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }
    const removeBox = (id) =>{
        setBoxes(boxes => boxes.filter(box => box.id !== id))
    }
    return(
        <div>
            <h3>Boxes!</h3>
            <NewBoxForm addBox={addBox}/>
            <div>
                {boxes.map(({id, width, height, backgroundColor}) => 
            <Box 
            id ={id}
            key={id}
            width={width}
            height={height}
            backgroundColor={backgroundColor}
            handleRemove={removeBox}
            />)}
            </div>
        </div>
    )
}

export default BoxList;