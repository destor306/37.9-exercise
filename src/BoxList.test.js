import React from 'react';
import {render, fireEvent} from "@testing-library/react"
import BoxList from './BoxList';


it('renders without crashing', ()=>{
    render(<BoxList />)
})

it('matches snapshot', ()=>{
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot();
})

it('should add new Box', ()=>{
    const{queryByText, getByLabelText, getByText } = render(<BoxList />)
    const widthInput = getByLabelText('width')
    const heightInput = getByLabelText('height')
    const colorInput = getByLabelText('color')
    const btn = queryByText('Add a box!');
    // Fill out form inputs
    fireEvent.change(widthInput, { target: { value: '5' } });
    fireEvent.change(heightInput, { target: { value: '5' } });
    fireEvent.change(colorInput, { target: { value: 'blue' } });

    // Submit form
    fireEvent.click(btn);

    const removeBtn = queryByText("Remove");
    expect(removeBtn).toBeInTheDocument();
    expect(removeBtn.previousSibling).toHaveStyle(`
        width: 5em;
        height: 5em;
        background-color: blue
    `)
})

it('should delete a Box', ()=>{
    const{queryByText, getByLabelText, getByText } = render(<BoxList />)
    const widthInput = getByLabelText('width')
    const heightInput = getByLabelText('height')
    const colorInput = getByLabelText('color')
    const btn = queryByText('Add a box!');
    // Fill out form inputs
    fireEvent.change(widthInput, { target: { value: '5' } });
    fireEvent.change(heightInput, { target: { value: '5' } });
    fireEvent.change(colorInput, { target: { value: 'blue' } });

    // Submit form
    fireEvent.click(btn);

    const removeBtn = queryByText("Remove");
    fireEvent.click(removeBtn);
    
    expect(removeBtn).not.toBeInTheDocument();
})