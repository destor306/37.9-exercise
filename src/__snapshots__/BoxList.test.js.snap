// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Boxes!
    </h3>
    <form>
      <label
        for="width"
      >
        width
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        height
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="color"
      >
        color
      </label>
      <input
        id="color"
        name="backgroundColor"
        type="text"
        value=""
      />
      <button>
        Add a box!
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
